File: gamslib/csp/csp-scalar.py
#  MIP written by GAMS Convert at 12/13/18 11:24:22
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         10        6        0        4        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         20        1       19        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         47       47        0        0


from pyomo.environ import *

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(None,None),initialize=0)
m.b2 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b3 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b4 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b5 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b6 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b7 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b8 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b9 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b10 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b11 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b12 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b13 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b14 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b15 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b16 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b17 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b18 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b19 = Var(within=Binary,bounds=(0,1),initialize=0)
m.b20 = Var(within=Binary,bounds=(0,1),initialize=0)

m.obj = Objective(expr=m.x1, sense=minimize)

m.c1 = Constraint(expr=   m.b2 + m.b3 + m.b4 == 1)

m.c2 = Constraint(expr=   m.b5 + m.b6 == 1)

m.c3 = Constraint(expr=   m.b7 + m.b8 + m.b9 == 1)

m.c4 = Constraint(expr=   m.b10 + m.b11 + m.b12 == 1)

m.c5 = Constraint(expr=   m.b13 + m.b14 + m.b15 + m.b16 == 1)

m.c6 = Constraint(expr=   m.b17 + m.b18 + m.b19 + m.b20 == 1)

m.c7 = Constraint(expr= - m.x1 - m.b2 - m.b6 - m.b8 - m.b10 - m.b14 - m.b20 <= -6)

m.c8 = Constraint(expr= - m.x1 - m.b4 - m.b5 - m.b7 - m.b12 - m.b13 - m.b19 <= -6)

m.c9 = Constraint(expr= - m.x1 - m.b3 - m.b5 - m.b9 - m.b11 - m.b15 - m.b17 <= -6)

m.c10 = Constraint(expr= - m.x1 - m.b4 - m.b5 - m.b7 - m.b12 - m.b16 - m.b18 <= -6)

File: gamslib/csp/csp-scalar.jl
#  MIP written by GAMS Convert at 10/11/20 13:00:58
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         10        6        0        4        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         20        1       19        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         47       47        0        0


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, x1, start=0)
@variable(m, 0 <= b2 <= 1, binary=true, start=0)
@variable(m, 0 <= b3 <= 1, binary=true, start=0)
@variable(m, 0 <= b4 <= 1, binary=true, start=0)
@variable(m, 0 <= b5 <= 1, binary=true, start=0)
@variable(m, 0 <= b6 <= 1, binary=true, start=0)
@variable(m, 0 <= b7 <= 1, binary=true, start=0)
@variable(m, 0 <= b8 <= 1, binary=true, start=0)
@variable(m, 0 <= b9 <= 1, binary=true, start=0)
@variable(m, 0 <= b10 <= 1, binary=true, start=0)
@variable(m, 0 <= b11 <= 1, binary=true, start=0)
@variable(m, 0 <= b12 <= 1, binary=true, start=0)
@variable(m, 0 <= b13 <= 1, binary=true, start=0)
@variable(m, 0 <= b14 <= 1, binary=true, start=0)
@variable(m, 0 <= b15 <= 1, binary=true, start=0)
@variable(m, 0 <= b16 <= 1, binary=true, start=0)
@variable(m, 0 <= b17 <= 1, binary=true, start=0)
@variable(m, 0 <= b18 <= 1, binary=true, start=0)
@variable(m, 0 <= b19 <= 1, binary=true, start=0)
@variable(m, 0 <= b20 <= 1, binary=true, start=0)

@objective(m, Min, x1)

@constraint(m, b2 + b3 + b4 == 1)

@constraint(m, b5 + b6 == 1)

@constraint(m, b7 + b8 + b9 == 1)

@constraint(m, b10 + b11 + b12 == 1)

@constraint(m, b13 + b14 + b15 + b16 == 1)

@constraint(m, b17 + b18 + b19 + b20 == 1)

@constraint(m,  - x1 - b2 - b6 - b8 - b10 - b14 - b20 <= -6)

@constraint(m,  - x1 - b4 - b5 - b7 - b12 - b13 - b19 <= -6)

@constraint(m,  - x1 - b3 - b5 - b9 - b11 - b15 - b17 <= -6)

@constraint(m,  - x1 - b4 - b5 - b7 - b12 - b16 - b18 <= -6)
